packages/core/src/services/chains/run.ts

```diff
@@ -147,26 +178,31 @@ async function iterate({
       output: step.output,
     }).then((r) => r.unwrap())
 
-    const aiStreamConsumedResult = await consumeStream({
+    const consumedStream = await consumeStream({
       controller,
       result: aiResult,
     })
-    const response = await providerProcessor.call({
-      aiResult,
-      streamConsumedResult: aiStreamConsumedResult,
-      startTime: stepStartTime,
-    })
+    const response = await providerProcessor
+      .call({
+        aiResult,
+        finishReason: consumedStream.finishReason,
+        startTime: stepStartTime,
+      })
+      .then((r) => r.unwrap())
```

Comment:
Now if the AI respond with a chunks that are of `type: 'error'` or { type: 'finish', finishReason: 'error'}` we throw an error that is persisted in the errorable